pkg/controller/node/nodecontroller.go

```diff
@@ -496,6 +488,43 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
+var nodeExistsInCloudProvider = func(nc *NodeController, nodeName string) (bool, error) {
```

Comment:
I did it this way for testability. I could also made it a function pointer in the struct. I lean towards stubbable package functions for testing, but I'll change it if you'd like.